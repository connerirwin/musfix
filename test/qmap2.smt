; Formal parameter names can collide in qualifiers
(qualif Less ((a int) (b int))   (< a b))
(qualif Or   ((a bool)(b bool)) (|| a b))

; And also in the wf constaints
(wf $k0 ((v0 int) (v1 int)))
(wf $k1 ((v0 bool)(v1 bool)))
