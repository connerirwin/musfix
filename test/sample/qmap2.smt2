; Formal parameter names can collide in qualifiers
(qualif Less ((a Int) (b Int))   (< a b))
(qualif Or   ((a Bool)(b Bool)) (|| a b))

; And also in the wf constaints
(wf $k0 ((v0 Int) (v1 Int)))
(wf $k1 ((v0 Bool)(v1 Bool)))
