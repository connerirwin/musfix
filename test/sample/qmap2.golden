
Preparing to find greatest fixpoint...

Inputs
--------
Reading from file test/sample/qmap2.smt2
Qualifier "Less" [Var IntS "a",Var IntS "b"] (Binary Lt (Var IntS "a") (Var IntS "b"))
Qualifier "Or" [Var BoolS "a",Var BoolS "b"] (Binary Or (Var BoolS "a") (Var BoolS "b"))
WFConstraint "$k0" [Var IntS "v0",Var IntS "v1"]
WFConstraint "$k1" [Var BoolS "v0",Var BoolS "v1"]

QMAP
--------
fromList [("$k0",QSpace {_qualifiers = [Binary Lt (Var IntS "v0") (Var IntS "v1"),Binary Lt (Var IntS "v1") (Var IntS "v0")], _maxCount = 2}),("$k1",QSpace {_qualifiers = [Binary Or (Var BoolS "v0") (Var BoolS "v1"),Binary Or (Var BoolS "v1") (Var BoolS "v0")], _maxCount = 2})]

Candidates
--------
[Candidate {solution = fromList [("$k0",fromList []),("$k1",fromList [])], validConstraints = fromList [], invalidConstraints = fromList [], label = "0"}]


Final candidates: 
0: [$k0 -> {}, $k1 -> {}] (0 0)

